### PR TITLE
CHORE: Migrate from Parcel to Vite

### PR DESCRIPTION
**Pull request by KOOKIIEStudios**
## Abstract
Repository previously relied on Parcel, which has been deprecated as of OCT 2022.
This is a problem as Parcel relies on outdated dependencies that are known security risks.
Migration to Vite is crucial for maintaining continued support.

## Pre-merge Testing
Passed cursory testing to allow previews and builds.

GTS/ESLint will not work due to lack of TS 4.5+ support for ESLint.

### CODE REVIEW

File: client/src/main.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ const game = {
 	type: Phaser.AUTO,
 	width: 1280, // TODO: Figure out how big this game should be
 	height: 720,
-	parent: 'moogiwara',
+	parent: 'MOOgiwara',
```
Comment (Bratah123) [inline client/src/main.ts:10]: I think for any internal usage of phaser naming, we will be using all lowercase naming

**Review by Bratah123 — COMMENTED** [review]
One small nit, other than that LGTM

File: client/src/main.ts (hunk 1 above)
Comment (KOOKIIEStudios) [reply client/src/main.ts:10]: I was under the assumption that this is referring to the HTML tag parent
is this a module name?

Comment (KOOKIIEStudios) [reply client/src/main.ts:10]: i don't actually know what this `parent` is

**Pull request merged**